### PR TITLE
Implement pi-distance-class

### PR DESCRIPTION
**Pull request by SevKrok**

### CODE REVIEW

**Review by RabenkoYevhenii — APPROVED** [review]
But everything else is perfect ;)

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,63 @@
...
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
```
[the remark this answers is not in the document]
Comment (SevKrok) [reply app/main.py:17]: I'm playing it safe)))

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,63 @@
...
+        if isinstance(other, Distance):
+            return self.km <= other.km
+        elif isinstance(other, (int, float)):
+            return self.km <= other
+
+    def __ge__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, Distance):
+            return self.km >= other.km
+        elif isinstance(other, (int, float)):
+            return self.km >= other
```
Comment (fsocie7y) [inline app/main.py:63]: try to rewrite these two methods in one line each. Use previously created magic methods
Hint: <= is the same as not >

File: app/main.py [hunk 3]
```diff
@@ -1,3 +1,63 @@
...
+        if isinstance(other, Distance):
+            return self.km > other.km
+        elif isinstance(other, (int, float)):
+            return self.km > other
+
+    def __eq__(self, other: int | float | Distance) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        elif isinstance(other, (int, float)):
+            return self.km == other
```
Comment (fsocie7y) [inline app/main.py:51]: You can skip second check here and directly return from this point
in one case above too

File: app/main.py (hunk 1 above)
Comment (fsocie7y) [inline app/main.py:17]: here, as Yevhenii gives advice

Comment (fsocie7y) [inline app/main.py:17]: here, as Yevhenii gives advice

Comment (fsocie7y) [inline app/main.py:17]: here, as Yevhenii gives advice

Comment (fsocie7y) [inline app/main.py:17]: here, as Yevhenii gives advice

**Review by fsocie7y — CHANGES_REQUESTED** [review]

**Review by fsocie7y — APPROVED** [review]
GJ 🥇

**Pull request closed without merging**